editoast/src/views/timetable.rs

```diff
@@ -59,25 +67,111 @@ async fn get_conflicts(
         Some(timetable) => timetable,
         None => return Err(TimetableError::NotFound { timetable_id }.into()),
     };
-    let timetable_with_schedules = timetable.with_train_schedules(db_pool).await?;
+    let timetable_with_schedules = timetable.with_train_schedules(db_pool.clone()).await?;
 
-    if timetable_with_schedules.train_schedules.len() < 2 {
-        return Ok(Json(vec![]));
+    let mut simulation_outputs = vec![];
+    for train_schedule in timetable_with_schedules.train_schedules.iter() {
+        let db_pool2 = db_pool.clone();
+        let train_schedule_id = train_schedule.id;
+        simulation_outputs.push(
+            block::<_, Result<_>>(move || {
+                use crate::tables::osrd_infra_simulationoutput::dsl;
+                use diesel::prelude::*;
+                let mut conn = db_pool2.get()?;
+                let mut sim_outputs: Vec<SimulationOutput> = dsl::osrd_infra_simulationoutput
+                    .filter(dsl::train_schedule_id.eq(train_schedule_id))
+                    .load(&mut conn)?;
+                if sim_outputs.is_empty() {
+                    return Err(
+                        TrainScheduleError::UnsimulatedTrainSchedule { train_schedule_id }.into(),
+                    );
+                }
+                assert!(sim_outputs.len() == 1);
+                Ok(sim_outputs.remove(0))
+            })
+            .await
+            .unwrap()?, // FIXME: maybe this should be converted to an editoast error?
+        );
```

Comment:
this loop runs N sql requests, when N is the number of trains in the timetable; that's not great, can you run a single request instead?